crates/fpe/src/ffi/float.rs

```diff
@@ -0,0 +1,75 @@
+use std::ffi::{c_char, c_double, c_int};
+
+use cosmian_ffi_utils::{ffi_read_bytes, ffi_unwrap};
+
+use crate::core::Float;
+
+unsafe extern "C" fn fpe(
+    output: *mut c_double,
+    input: c_double,
+    key_ptr: *const c_char,
+    key_len: c_int,
+    tweak_ptr: *const c_char,
+    tweak_len: c_int,
+    encrypt_flag: bool,
+) -> c_int {
+    let key_bytes = ffi_read_bytes!("key", key_ptr, key_len);
+    let tweak_bytes = ffi_read_bytes!("tweak", tweak_ptr, tweak_len);
+
+    // Copy the contents of the slice into the 32-array
+    let mut key: [u8; 32] = [0; 32];
+    key.copy_from_slice(key_bytes);
```

Comment:
I think there is a `TryFrom` implementation to do that… (and throw an error if the size is not correct)

Reply:
thanks